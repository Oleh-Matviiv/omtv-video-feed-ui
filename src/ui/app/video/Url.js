import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Play = styled.video.attrs({
  className: 'video-play',
})`
  grid-area: play;
`;

const Url = ({ src }) => (
  <Play width="100%" src={ src } controls>
    Your browser doesn't support HTML5 video.
  </Play>
);
Url.propTypes = {
  src: PropTypes.string.isRequired,
};

export default Url;
