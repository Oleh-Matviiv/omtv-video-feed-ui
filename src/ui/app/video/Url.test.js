import React from 'react';
import { mount } from 'enzyme';
import 'jest-styled-components';

import Url from './Url';

describe('snapshots', () => {
  it('Title', () => {
    const wrapper = mount(
      <Url src="http://g.com/1.mp4" />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
