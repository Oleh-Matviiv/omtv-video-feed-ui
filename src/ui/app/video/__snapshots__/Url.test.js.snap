// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots Title 1`] = `
.c0 {
  grid-area: play;
}

<Url
  src="http://g.com/1.mp4"
>
  <styled.video
    controls={true}
    src="http://g.com/1.mp4"
    width="100%"
  >
    <StyledComponent
      controls={true}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [
            Object {
              "className": "video-play",
            },
          ],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  grid-area: play;
",
            ],
          },
          "displayName": "styled.video",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "video",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      src="http://g.com/1.mp4"
      width="100%"
    >
      <video
        className="video-play c0"
        controls={true}
        src="http://g.com/1.mp4"
        width="100%"
      >
        Your browser doesn't support HTML5 video.
      </video>
    </StyledComponent>
  </styled.video>
</Url>
`;
